pkg/antctl/raw/traceflow/command.go

```diff
@@ -233,90 +216,71 @@ func newTraceflow(client kubernetes.Interface) (*v1alpha1.Traceflow, error) {
+	fields, err := getPortFields(cleanFlow)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return false, nil
-		}
-		return false, fmt.Errorf("failed to get Pod from Kubernetes API: %w", err)
+		return nil, fmt.Errorf("error when parsing the flow: %w", err)
 	}
-	return true, nil
-}
 
-func parseFlow() (*v1alpha1.Packet, error) {
-	cleanFlow := strings.ReplaceAll(option.flow, " ", "")
 	pkt := new(v1alpha1.Packet)
-
-	for _, v := range strings.Split(cleanFlow, ",") {
-		n, ok := protocols[v]
-		if ok {
-			(*pkt).IPHeader.Protocol = n
+	for k, v := range protocols {
+		if _, ok := fields[k]; ok {
+			pkt.IPHeader.Protocol = v
 			break
 		}
 	}
 
-	r, ok, err := getFieldPortValue(cleanFlow, "tcp_src")
-	if err != nil {
-		return nil, fmt.Errorf("error when get tcp_src value: %w", err)
-	}
-	if ok {
-		if (*pkt).TransportHeader.TCP == nil {
-			(*pkt).TransportHeader.TCP = new(v1alpha1.TCPHeader)
+	if r, ok := fields["tcp_src"]; ok {
+		if pkt.TransportHeader.TCP == nil {
+			pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
 		}
-		(*pkt).TransportHeader.TCP.SrcPort = int32(r)
-	}
-	r, ok, err = getFieldPortValue(cleanFlow, "tcp_dst")
-	if err != nil {
-		return nil, fmt.Errorf("error when get tcp_dst value: %w", err)
+		pkt.TransportHeader.TCP.SrcPort = int32(r)
 	}
-	if ok {
-		if (*pkt).TransportHeader.TCP == nil {
-			(*pkt).TransportHeader.TCP = new(v1alpha1.TCPHeader)
+	if r, ok := fields["tcp_dst"]; ok {
+		if pkt.TransportHeader.TCP == nil {
+			pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
 		}
-		(*pkt).TransportHeader.TCP.DstPort = int32(r)
+		pkt.TransportHeader.TCP.SrcPort = int32(r)
 	}
-	r, ok, err = getFieldPortValue(cleanFlow, "udp_src")
-	if err != nil {
-		return nil, fmt.Errorf("error when get udp_src value: %w", err)
-	}
-	if ok {
-		if (*pkt).TransportHeader.UDP == nil {
-			(*pkt).TransportHeader.UDP = new(v1alpha1.UDPHeader)
+	if r, ok := fields["udp_src"]; ok {
+		if pkt.TransportHeader.TCP == nil {
+			pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
 		}
-		(*pkt).TransportHeader.UDP.SrcPort = int32(r)
-	}
-	r, ok, err = getFieldPortValue(cleanFlow, "udp_dst")
-	if err != nil {
-		return nil, fmt.Errorf("error when get udp_dst value: %w", err)
+		pkt.TransportHeader.TCP.SrcPort = int32(r)
 	}
-	if ok {
-		if (*pkt).TransportHeader.UDP == nil {
-			(*pkt).TransportHeader.UDP = new(v1alpha1.UDPHeader)
+	if r, ok := fields["udp_dst"]; ok {
+		if pkt.TransportHeader.TCP == nil {
+			pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
 		}
-		(*pkt).TransportHeader.UDP.DstPort = int32(r)
+		pkt.TransportHeader.TCP.SrcPort = int32(r)
 	}
 
 	return pkt, nil
 }
 
-func getFieldPortValue(cleanFlow string, f string) (int, bool, error) {
+func getPortFields(cleanFlow string) (map[string]int, error) {
+	fields := map[string]int{}
 	for _, v := range strings.Split(cleanFlow, ",") {
-		m, err := regexp.MatchString(fmt.Sprintf("%s=[0-9]+", f), v)
+		ok, err := regexp.MatchString(".+=[0-9]+", v)
```

Comment:
lint has failed on this `calling regexp.MatchString in a loop has poor performance, consider using regexp.Compile.`
But I think regexp match is not need at all. You can just split `v` with `=`: if it has two items, then `strconv.Atoi` will check if the second item is a number; if it has only one item, set `fields[v] = 0` like L280

Reply:
Good idea. Updated.